klayout/drc/rule_decks/lres.drc

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+################################################################################################
+# Copyright 2022 GlobalFoundries PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################################
+
+if FEOL
+  #================================================
+  #----------------N+ POLY RESISTOR----------------
+  #================================================
+
+  logger.info('Starting LRES derivations')
+  lres_poly = poly2.and(nplus).interacting(sab).interacting(res_mk)
+
+  # Rule LRES.1: Minimum width of Poly2 resistor. is 0.8µm
+  logger.info('Executing rule LRES.1')
+  lres1_l1 = lres_poly.width(0.8.um, euclidian)
+  lres1_l1.output('LRES.1', 'LRES.1 : Minimum width of Poly2 resistor. : 0.8µm')
+  lres1_l1.forget
+
+  # Rule LRES.2: Minimum space between Poly2 resistors. is 0.4µm
+  logger.info('Executing rule LRES.2')
+  lres2_l1 = lres_poly.isolated(0.4.um, euclidian)
+  lres2_l1.output('LRES.2', 'LRES.2 : Minimum space between Poly2 resistors. : 0.4µm')
+  lres2_l1.forget
+
+  # Rule LRES.3: Minimum space from Poly2 resistor to COMP.
+  logger.info('Executing rule LRES.3')
+  lres3_l1 = lres_poly.separation(comp, 0.6.um, euclidian).polygons(0.001.um).or(comp.not_outside(lres_poly))
```

Comment:
why do we have .001 here?

Reply:
All `polygons(0.001.um)` are for visibility and being able to `join` the results. But we can't remove that as we combine the layers down the line as well.

Reply:
can you clarify what "All polygons(0.001.um) are for visibility" means? (or even better add a comment in the rule that explain that)

Reply:
When run any kind of spacing check, it generates edges rather than polygons. We convert them to polygons with with `0.001.um`. This is what this does.

Reply:
and what is the reason we don't keep them as edges? is to be able to do `or()` on them or to be able to call `output()`?

Reply:
would https://www.klayout.de/doc/code/class_EdgeProcessor.html#method43 also work here? what would be the trade off of using one versus the other?

Reply:
filed #67 to discuss this further